### PR TITLE
fix: comparing wei with bytes32

### LINKED ISSUE #43: comparing BaseUInt256Value and Bytes32 issue
Wei is extending BaseUInt256Value.

We have: 
`Wei.ZERO.equals(Bytes32.ZERO)) = false`
`Bytes32.ZERO.equals(Wei.Zero) = true`

We would like to have:
`Wei.ZERO.equals(Bytes32.ZERO)) = true`

### PR DESCRIPTION
**Pull request by letypequividelespoubelles**
## Fixed Issue(s)
fixes #43

### CODE REVIEW

**Comment by letypequividelespoubelles** [comment]
I have read the CLA Document and I hereby sign the CLA

File: units/src/test/java/org/apache/tuweni/units/ethereum/WeiTest.java [hunk 1]
```diff
@@ -79,4 +80,10 @@ void testToLong() {
   void toBigIntegerIsPositive() {
     assertEquals(1, Wei.valueOf(UInt256.MAX_VALUE).toBigInteger().signum());
   }
+
+  @Test
+  void compareWeiWithBytes32() {
+    assertEquals(Bytes32.ZERO, Wei.valueOf(0));
+    assertEquals(Wei.valueOf(0), Bytes32.ZERO);
```
Comment (fluentcrafter) [inline units/src/test/java/org/apache/tuweni/units/ethereum/WeiTest.java:87]: IMO it should also test other values than Zero...

Comment (ahamlat) [reply units/src/test/java/org/apache/tuweni/units/ethereum/WeiTest.java:87]: +1

File: units/src/main/java/org/apache/tuweni/units/bigints/BaseUInt256Value.java [hunk 2]
```diff
@@ -338,12 +338,12 @@ public boolean equals(Object obj) {
...
     }
-    if (obj instanceof Bytes) {
-      final Bytes other = (Bytes) obj;
-      return this.value.equals(other);
+    if (!(obj instanceof Bytes32)) {
+      return false;
     }
 
-    return false;
+    Bytes32 other = (Bytes32) obj;
```
Comment (ahamlat) [inline units/src/main/java/org/apache/tuweni/units/bigints/BaseUInt256Value.java:345]: IMO, you don't need to cast, you just delegate the work to UInt256. I wonder if it is not even better to do directly
return this.value.equals(other); as the UInt256 implementation does the same checks.

**Review by fluentcrafter — APPROVED** [review]
